### ISSUE TITLE
Recall last argument from previous command

### ISSUE THREAD

**Issue by amitkot** [xonsh/xonsh#1480]
I'm looking for a way to recall the last argument of the previous command, similar to `Alt+.` on Bash.
I'm using vi mode, so `Esc+_` does not work.
Is there a way to get this on Xonsh?

Posted this on the IRC channel a couple of weeks ago and got no reply, so I assume this is currently not implemented.

**Comment by gforsyth** [comment]
Hey @amitkot, thanks for opening this.  There is not currently a shortcut for doing this, although a naive implementation wouldn't be too hard at this point.  
In the interim, you could do something like this as a temporary workaround:

```console
gil@bad_cat ~ $ aliases['lc'] = lambda args, stdin=None: _.inp.split()[-1]
gil@bad_cat ~ $ ls tmp
file1  file2  file3
gil@bad_cat ~ $ cd $(lc)
gil@bad_cat ~/tmp $ 
```

where `_.inp.split()[-1]` will return the last argument of the previous command

**Comment by Cadair** [comment]
Where in the code base would a key binding like this belong?

**Comment by scopatz** [comment]
@Cadair - It would go in the PTK shell key bindings  (`xonsh.ptk.key_bindings`) or in the readline startup
`xonsh.readline_shell.setup_readline()`

**Comment by Cadair** [comment]
This seems to have been implemented?

**Comment by scopatz** [comment]
It might have been. @gforsyth  would know more ;)

**Comment by duerrp** [comment]
Something like this should do the trick right? I guess one should also be able to implement Alt + m like this (another one that's hard to erase from muscle memory...)

```
from prompt_toolkit.keys import Keys
from prompt_toolkit.filters import Condition, EmacsInsertMode, ViInsertMode

@events.on_ptk_create
def custom_keybindings(bindings, **kw):
    handler = bindings.registry.add_binding

    @handler(Keys.Escape, '.')
    def say_hi(event):
        event.current_buffer.insert_text(_.inp.split()[-1])
```

**Comment by astronouth7303** [comment]
@duerrp Looks about right, although I will note that `_` is not always the last command. (It can be the result of a Python expression, or if it's the first line, nothing at all.)

**Comment by gforsyth** [comment]
I wonder if this should go into the bashisms xontrib?

**Comment by duerrp** [comment]
At the current state, it's hardly more than a cheap hack (e.g., it will throw an exception at the first line) and it does not have the power of the real Alt .. Moreover, I extensively use Alt . and Alt m in zsh, so bashisms does not sound like the right place :-)

**Comment by gforsyth** [comment]
See #2310 -- I added some conditions so it doesn't barf and removed the dependence on `_`, which as @astronouth7303 pointed out isn't completely robust.

**Comment by gforsyth** [comment]
Implemented in #2310
